### PR TITLE
GNA add InsertTransposeBeforeMatmul unit tests

### PR DESCRIPTION
**Pull request by evkotov**
### Details:
 [GNA] add ngraph-based transformations tests
- all new tests will be put into gnaFuncTests executable binary
- create dir  inference-engine/tests/unit/gna/ngraph/transformations for all ngraph-based transformations tests
- add test on InsertTransposeBeforeMatmul transformation
- fix InsertTransposeBeforeMatmul transformation copying runtime info

### Tickets:
 - 57583

### CODE REVIEW

**Review by elilobanova — APPROVED** [review]

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                                                   const ngraph::Shape & matmul_shape);
+    std::shared_ptr<ngraph::Function> get_reference(const ngraph::PartialShape & input_shape);
+public:
+    std::shared_ptr<ngraph::Function> function, reference_function;
+};
+
+void InsertTransposeBeforeMatmulTestFixture::SetUp() {
+    const auto& input_shape = std::get<0>(GetParam());
+    const auto& reshape_shape = std::get<1>(GetParam());
+    const auto& matmul_shape = std::get<2>(GetParam());
```
Comment (mryzhov) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:95]: jfyi, the following code mae be used in future:

const auto& input_shape, reshape_shape, matmul_shape;
std::tie(input_shape , reshape_shape, matmul_shape) = this->GetParam();

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    reference_func = createFunction(ngraph::PartialShape{2, 9}, ngraph::Shape{9, 2}, ngraph::Shape{2, 1});
+
+    const FunctionsComparator func_comparator = FunctionsComparator::with_default().enable(FunctionsComparator::ATTRIBUTES);
+    const FunctionsComparator::Result result = func_comparator(func, reference_func);
+    ASSERT_TRUE(result.valid);
+}
+
+TEST(TransformationTests, InsertTransposeBeforeMatmulTestReshapeInOutEq) {
```
Comment (mryzhov) [inline inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:57]: Looks like we can use TEST_P for negative cases here and do not duplicate the test logic twice.

File: inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp (hunk 1 above)
Comment (evkotov) [reply inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:95]: may be but we will get one long line, which will be less readable

Comment (evkotov) [reply inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:95]: error: declaration of ‘const auto& input_shape’ has no initializer

Comment (evkotov) [reply inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:95]: may be const auto& [input_shape, reshape_shape, matmul_shape] = this->GetParam(); ?

Comment (mryzhov) [reply inference-engine/tests/unit/gna/ngraph/transformations/gna_insert_transpose_before_matmul.cpp:95]: yes, yes, I've just copied your code) but this is just a possible way, I don't mind leaving it as is.

**Review by mryzhov — APPROVED** [review]

**Pull request merged**